pkg/distributor/distributor.go

```diff
@@ -290,18 +303,31 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 	var lastPartialErr error
 	removeReplica := false
 
+	numSamples := 0
+	for _, ts := range req.Timeseries {
+		numSamples += len(ts.Samples)
+	}
+	// Count the total samples in, prior to validation or deuplication, for comparison with other metrics.
+	incomingSamples.WithLabelValues(userID).Add(float64(numSamples))
+
 	if d.cfg.EnableHATracker && d.limits.AcceptHASamples(userID) && len(req.Timeseries) > 0 {
-		removeReplica, err = d.checkSample(ctx, userID, req.Timeseries[0])
+		cluster, replica := findHALabels(d.limits.HAReplicaLabel(userID), d.limits.HAClusterLabel(userID), req.Timeseries[0].Labels)
+		removeReplica, err = d.checkSample(ctx, userID, cluster, replica)
 		if err != nil {
+			if resp, ok := httpgrpc.HTTPResponseFromError(err); ok && resp.GetCode() == 202 {
+				// These samples have been deduped.
+				dedupedSamples.WithLabelValues(userID, cluster).Add(float64(numSamples))
```

Comment:
Would this be clearer and less code to move it into `checkSample`?

Reply:
We'd need to pass `numSamples` into checkSample, and we'll always have to check for a response code of 202. If you feel strongly about moving this I'll change it.

Reply:
Oh yes I see.  Now this is only checked in one place, this is fine.